### PR TITLE
Fix receiver names in legacyProvider

### PR DESCRIPTION
**Pull request by togettoyou**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Change receiver name from `c` to `p` in `NewRESTStorage` method

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by togettoyou** [comment]
Hi, @justinsb and @caesarxuchao, could you please assist me with reviewing?

File: pkg/registry/core/rest/storage_core.go [hunk 1]
```diff
@@ -147,13 +147,13 @@ func New(c Config) (*legacyProvider, error) {
 	return p, nil
 }
 
-func (c *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, error) {
-	apiGroupInfo, err := c.GenericConfig.NewRESTStorage(apiResourceConfigSource, restOptionsGetter)
+func (p *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, error) {
```
Comment (shivanshu1333) [inline pkg/registry/core/rest/storage_core.go:150]: LGTM, can you crosscheck if all the receivers with `legacyProvider` type are now changed to `p`

Comment (togettoyou) [reply pkg/registry/core/rest/storage_core.go:150]: Yes, I have confirmed that all the receivers with the `legacyProvider` type have been changed to `p`.

**Comment by togettoyou** [comment]
ping @deads2k @smarterclayton @wojtek-t @liggitt for approval

**Pull request merged**